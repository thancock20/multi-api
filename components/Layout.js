import Head from 'next/head'
import Header from './Header'

const Layout = ({ children, title }) => (
  <div>
    <Head>
      <title>{ title ? `Multi-API | ${title}` : 'Multi-API' }</title>
      <meta charset='utf8' />
      <meta name='viewport' content='initial-scale=1, width=device-width' />
    </Head>
    <Header />
    { children }
  </div>
)

export default Layout
