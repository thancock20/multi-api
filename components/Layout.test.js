import Layout from './Layout';

describe('Layout component', () => {
  it('renders with children and title', () => {
    const wrapper = shallow(
      <Layout title="Here's the title">Here are the children.</Layout>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with children and no title', () => {
    const wrapper = shallow(<Layout>Here are the children.</Layout>);
    expect(wrapper).toMatchSnapshot();
  });
});
