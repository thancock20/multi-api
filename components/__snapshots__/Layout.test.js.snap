// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders with children and no title 1`] = `
<div
  className="jsx-2205840965 container"
>
  <Head>
    <title
      className="jsx-2205840965"
    >
      Multi-API
    </title>
    <meta
      charSet="utf8"
      className="jsx-2205840965"
    />
    <meta
      className="jsx-2205840965"
      content="initial-scale=1, width=device-width"
      name="viewport"
    />
    <link
      className="jsx-2205840965"
      href="https://fonts.googleapis.com/css?family=Source+Code+Pro|Source+Sans+Pro|Source+Serif+Pro"
      rel="stylesheet"
    />
  </Head>
  <Header
    title="Multi-API"
  />
  Here are the children.
  <JSXStyle
    id="2205840965"
  >
    *{box-sizing:border-box;}html{font-size:1.5vmax;font-family:'Source Sans Pro',sans-serif;line-height:1.42857143;}body{color:#333;background-color:#fff;}h2{font-family:'Source Serif Pro',serif;font-size:2rem;margin-bottom:0;}.container{width:70vw;margin-left:15vw;}
  </JSXStyle>
</div>
`;

exports[`Layout component renders with children and title 1`] = `
<div
  className="jsx-2205840965 container"
>
  <Head>
    <title
      className="jsx-2205840965"
    >
      Multi-API | Here's the title
    </title>
    <meta
      charSet="utf8"
      className="jsx-2205840965"
    />
    <meta
      className="jsx-2205840965"
      content="initial-scale=1, width=device-width"
      name="viewport"
    />
    <link
      className="jsx-2205840965"
      href="https://fonts.googleapis.com/css?family=Source+Code+Pro|Source+Sans+Pro|Source+Serif+Pro"
      rel="stylesheet"
    />
  </Head>
  <Header
    title="Here's the title"
  />
  Here are the children.
  <JSXStyle
    id="2205840965"
  >
    *{box-sizing:border-box;}html{font-size:1.5vmax;font-family:'Source Sans Pro',sans-serif;line-height:1.42857143;}body{color:#333;background-color:#fff;}h2{font-family:'Source Serif Pro',serif;font-size:2rem;margin-bottom:0;}.container{width:70vw;margin-left:15vw;}
  </JSXStyle>
</div>
`;
